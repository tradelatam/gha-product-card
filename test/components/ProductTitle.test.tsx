//import * as React from 'react';

import React  from 'react';
import renderer  from 'react-test-renderer';
import { ProductCard } from '../../src/components/ProductCard';
import { ProductTitle }  from '../../src/components/ProductTitle';
import { product1 } from './data/products';


describe('Pruebas en ProductTitle', () => {  

    test('debe de mostrar el componente correctamente con el titulo personalizado', () => {  
        
        const wrapper = renderer.create(
            <ProductTitle title="Custom Product" className="custom-class" />
        )
        //console.log( wrapper.toJSON() );
        expect( wrapper.toJSON() ).toMatchSnapshot();
        
    });

    test('debe de mostrar el componente con el nombre del producto', () => {  

        const wrapper = renderer.create(
            <ProductCard product={ product1 }>
                {
                    () => (
                        <ProductTitle />
                    )
                }
            </ProductCard>
        );

        expect(wrapper.toJSON()).toMatchSnapshot();

    });
})



// describe('it', () => {
//   it('renders without crashing', () => {
//     const div = document.createElement('div');
//     //ReactDOM.render(<Thing />, div);
//     ReactDOM.unmountComponentAtNode(div);
//   });
// });

