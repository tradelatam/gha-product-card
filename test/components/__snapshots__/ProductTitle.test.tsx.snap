// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en ProductTitle debe de mostrar el componente con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="undefined productDescription"
  >
    Coffee Mug - Sin imagen
  </span>
</div>
`;

exports[`Pruebas en ProductTitle debe de mostrar el componente correctamente con el titulo personalizado 1`] = `
<span
  className="custom-class productDescription"
>
  Custom Product
</span>
`;
